packages/api/server.js

```diff
@@ -38,99 +33,60 @@ global.headCollection = process.env.HEAD_COLLECTION || 'testing/main/repos';
-      dataset: client,
-      kind: 'express-sessions'
-    }),
-    secret: process.env.SESSION_SECRET || 'no secret',
-    resave: false,
-    saveUninitialized: true
-  })
-);
-
-app.get('/api/repos', async (req, res) => {
-  const repository = new Repository(null);
-  const result = await repository.getCollection('dummy-repositories');
-
-  const repoNames = [];
-
-  for (let index = 0; index < result.length; index++) {
-    const id = result[index].repositoryid;
-    repoNames.push(id);
-  }
-
-  const jsonObject = { repoNames: repoNames };
-  // TODO allow the requester to give search/filter criterion!
-  res
-    .status(200)
-    .send(jsonObject)
-    .end();
-});
 
-app.get('/api/auth', (req, res) => {
+app.delete('/api/repo/:orgname/:reponame/test/:testid', async (req, res) => {
+  const orgName = req.params.orgname;
+  const repoId = req.params.reponame;
+  const testName = req.params.testid;
+  const redirect = req.query.redirect;
   const state = v4();
-  req.session.authState = state;
-  const url = 'http://github.com/login/oauth/authorize?client_id=' + process.env.CLIENT_ID + '&state=' + req.session.authState + '&allow_signup=false';
+
+  await repo.storeTicket({
+    action: 'delete-test',
+    orgName: orgName,
+    repoId: repoId,
+    testName: testName,
+    state: state,
+    redirect: redirect
+  });
+
+  const url = 'http://github.com/login/oauth/authorize?client_id=' + process.env.CLIENT_ID + '&state=' + state + '&allow_signup=false';
   res.status(302).redirect(url);
 });
 
 app.get('/api/callback', async (req, res) => {
-  const redirect = process.env.FRONTEND_URL;
+  const ticket = repo.getTicket(req.param('state'));
+  const redirect = ticket.redirect || process.env.FRONTEND_URL;
 
-  if (req.param('state') !== req.session.authState) {
-    res.status(401).redirect(redirect);
+  if (ticket === null) {
+    res.status(404).redirect(redirect);
     return;
   }
 
-  const resp = await fetch('https://github.com/login/oauth/access_token', {
-    method: 'post',
-    body: JSON.stringify({
-      code: req.param('code'),
-      client_id: process.env.CLIENT_ID,
-      client_secret: process.env.CLIENT_SECRET,
-      state: req.session.authState
-    }),
-    headers: { 'Content-Type': 'application/json' }
-  });
-
-  const respText = await resp.text();
-  const queryObject = querystring.parse(respText);
+  const storedTicketState = ticket.state;
 
-  if (resp.status !== 200) {
+  if (req.param('state') !== storedTicketState) {
     res.status(401).redirect(redirect);
     return;
   }
 
-  const result = await fetch('https://api.github.com/user', {
-    method: 'get',
-    headers: { 'content-type': 'application/json', 'User-Agent': 'flaky.dev', Authorization: 'token ' + queryObject.access_token }
-  });
-
-  const resultJSON = await result.json();
+  const queryObject = await auth.retrieveAccessToken(req.param('code'), storedTicketState);
 
-  if (result.status !== 200) {
+  if (queryObject == null) {
```

Comment:
nit: tiny non blocking nit, I prefer `===` so that we're 100% sure of the variable's type.